# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/antitracking/StoragePrincipalHelper.cpp
# Commit: ddf4012a7652
# Full Hash: ddf4012a7652e36d144fc43bc99335276deeafbe
# Author: Christoph Kerschbaumer <ckerschb@christophkerschbaumer.com>
# Date: 2019-05-22 03:40:58
# Regressor Bug: 965637
# File Overlap Count: 1
# Description:
#   Bug 965637: Move CSP from Principal into Client, part 1: backend changes. r=mccr8
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D27654
# ==============================================================================

diff -r 6c2047bc2f3e -r ddf4012a7652 toolkit/components/antitracking/StoragePrincipalHelper.cpp
--- a/toolkit/components/antitracking/StoragePrincipalHelper.cpp	Tue May 21 22:51:54 2019 +0000
+++ b/toolkit/components/antitracking/StoragePrincipalHelper.cpp	Tue May 21 23:14:27 2019 +0000
@@ -116,23 +116,10 @@
     if (!spInfo.attrs().EqualsIgnoringFPD(pInfo.attrs()) ||
         spInfo.originNoSuffix() != pInfo.originNoSuffix() ||
         spInfo.spec() != pInfo.spec() || spInfo.domain() != pInfo.domain() ||
-        spInfo.baseDomain() != pInfo.baseDomain() ||
-        spInfo.securityPolicies().Length() !=
-            pInfo.securityPolicies().Length()) {
+        spInfo.baseDomain() != pInfo.baseDomain()) {
       return false;
     }
 
-    for (uint32_t i = 0; i < spInfo.securityPolicies().Length(); ++i) {
-      if (spInfo.securityPolicies()[i].policy() !=
-              pInfo.securityPolicies()[i].policy() ||
-          spInfo.securityPolicies()[i].reportOnlyFlag() !=
-              pInfo.securityPolicies()[i].reportOnlyFlag() ||
-          spInfo.securityPolicies()[i].deliveredViaMetaTagFlag() !=
-              pInfo.securityPolicies()[i].deliveredViaMetaTagFlag()) {
-        return false;
-      }
-    }
-
     return true;
   }
 